plugins/modules/foreman_os_default_template.py

```diff
@@ -126,35 +103,30 @@ def main():
             ['state', 'present', ['provisioning_template']],
             ['state', 'present_with_defaults', ['provisioning_template']],
         ),
+        name_map=name_map,
     )
 
-    (entity_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
     module.connect()
 
-    entity_dict['operatingsystem'] = find_operating_system_by_title(module, entity_dict['operatingsystem'])
-    entity_dict['template_kind'] = find_entities_by_name(
-        TemplateKind, [entity_dict['template_kind']], module)[0]
+    search = 'title~"{}"'.format(entity_dict['operatingsystem'])
+    entity_dict['operatingsystem'] = module.find_resource('operatingsystems', search, thin=True)
```

Comment:
Given my comment in https://github.com/theforeman/foreman-ansible-modules/pull/366#discussion_r306633638, I really think we should move that to a helper now ;)

Reply:
How about a fuzzy Flag?

Reply:
Sounds good

Reply:
I given my comment in https://github.com/theforeman/foreman-ansible-modules/pull/366#discussion_r306678268 on this too :)